pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -286,6 +286,16 @@ in rec {
           # lets have a clean always accessible version here.
           name = attrs.name or "${attrs.pname}-${attrs.version}";
 
+          # Nix only knows of `name` attribute and considers `pname` what parseDrvName
+          # spits out. Unfortunately, that fails to account for many real-life packages
+          # that have variants or dashes followed by number in their name.
+          # Since many of packages in nixpkgs already have pname corresponding
+          # to project name, we will export it here. We will fallback to the parseDrvName
+          # algorithm when pname is not present.
+          # We handle the version similarly.
+          pname = attrs.pname or (builtins.parseDrvName attrs.name).name;
+          version = attrs.version or ((x: if x != "" then x else null) (builtins.parseDrvName attrs.name).version);
```

Comment:
I agree with @volth that we should not have these fallbacks to `parseDrvName` because it is surprising to me that it would work like that. `pname` typically goes into `name`, so it is weird for `name` to go into `pname` in some cases.

Without these fallbacks the change seems quite uncontroversial. You are just saying make the attrs available in meta if they were passed in to mkDerivation.

Reply:
If we want to make something like `nameForRepology` in the website export code with this logic, we could do that.